backend/src/modules/transactions/transactions.controller.ts

```diff
@@ -0,0 +1,66 @@
+import { Controller, Get, Post, Delete, Body, Param, UsePipes } from '@nestjs/common';
+import { TransactionsService } from './transactions.service';
+import { CreateTransactionDto } from './dto/create-transaction.dto';
+import { Transactions } from './interfaces/transactions.interface';
+import { ValidationPipe } from '../../common/pipes/validation.pipe'
+
+@Controller('transactions')
+export class TransactionsController {
+
+  constructor(private readonly transactionsService: TransactionsService) {}
+
+  @Post()
+  @UsePipes(new ValidationPipe())
+  async create(@Body() createTransactionDto: CreateTransactionDto) {
+    console.log('Transaction received:');
+    console.log(CreateTransactionDto);
+    this.transactionsService.create(createTransactionDto);
+  }
+
+  @Delete(':id')
+  deleteTransaction(@Param('id') id: string): Promise<any> {
+    return this.transactionsService.remove(id);
+  }
+
+  @Get()
+  getAllTransactions(): Promise<Transactions[]> {
+    return this.transactionsService.findAll();
+  }
+
+  @Get('/income')
+  getIncome(): Promise<Transactions[]>{
+  	return this.transactionsService.findIncome();
+  }
+
+   @Get('/expenses')
+  getExpenses(): Promise<Transactions[]>{
+  	return this.transactionsService.findExpenses();
+  }
+
+  @Get(':id')
+  getTransaction(@Param('id') id: string): Promise<Transactions> {
+    return this.transactionsService.findById(id);
+  }
+
+  @Get('account/:account/:date/:number?')
+  getAccountTransactions(@Param('account') account: string, @Param('date') date: string, @Param('number') number: number)
+  : Promise<Transactions[]>
+  {
+  	return this.transactionsService.findAccountTransactions(account, date, number);
+  }
+
+  @Get('/expenses/account/:account/:date/:number?')
+  getAccountExpenses(@Param('account') account: string, @Param('date') date: string, @Param('number') number: number)
```

Comment:
paprastai path'e parametrai naudojami tik jei jie yra kaip direktorija, tarkim jei mes turėtume daugiau detalių apie transactionus, galima būtų daryt ':id/details', bet čia ne vieta tokiem `@Param`

Reply:
Ye, butu gerai pasiskaityti apie skirtumus tarp path variables ir query parameters, pvz: https://medium.com/@fullsour/when-should-you-use-path-variable-and-query-parameter-a346790e8a6d